ion-c-sys/src/writer.rs

```diff
@@ -383,4 +413,83 @@ impl Drop for IonCWriterHandle<'_> {
             ionc!(ion_writer_close(self.writer)).unwrap()
         }
     }
+}
+
+/// Context for writing fields in with proper reference lifetimes.
```

Comment:
I think you stated the problem in the chat group nicely. Can we get some of that context here? Specifically, that it ensures usages where both the field name and value are supplied without lifetime issues.

Reply:
Good point, let me add some additional color here about what the point of this thing is.